test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -1534,19 +1634,54 @@ public synchronized String startNode(Settings.Builder settings) {
      * Starts a node with the given settings and returns it's name.
      */
     public synchronized String startNode(Settings settings) {
-        NodeAndClient buildNode = buildNode(settings);
-        buildNode.startNode();
-        publishNode(buildNode);
+        final int defaultMinMasterNodes = getMinMasterNodes(getMasterNodesCount() + (Node.NODE_MASTER_SETTING.get(settings) ? 1 : 0));
+        NodeAndClient buildNode = buildNode(settings, defaultMinMasterNodes);
+        startAndPublishNodesAndClients(Collections.singletonList(buildNode));
         return buildNode.name;
     }
 
+    /**
+     * updates the min master nodes setting in the current running cluster.
+     *
+     * @param eligableMasterNodeCount the number of master eligable nodes to use as basis for the min master node setting
+     */
+    private int updateMinMasterNodes(int eligableMasterNodeCount) {
+        assert autoManageMinMasterNodes;
+        final int currentMasters = getMasterNodesCount();
+        final int minMasterNodes = getMinMasterNodes(eligableMasterNodeCount);
+        if (currentMasters == 0) {
```

Comment:
maybe we could simply do if (getMasterNodesCount>0) rather than having an empty if block?